src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionHelper.cs

```diff
@@ -38,85 +38,97 @@ public static string GetExpressionText(LambdaExpression expression, ExpressionTe
+                        containsIndexers = true;
+                        var binaryExpression = (BinaryExpression)part;
+                        var indexerInvokationResult = GetIndexerInvocation(
+                                binaryExpression.Right,
+                                parameters);
 
-                    nameParts.Push(
-                        GetIndexerInvocation(
-                            methodExpression.Arguments.Single(),
-                            expression.Parameters.ToArray()));
-
-                    part = methodExpression.Object;
-                }
-                else if (part.NodeType == ExpressionType.ArrayIndex)
-                {
-                    containsIndexers = true;
-                    var binaryExpression = (BinaryExpression)part;
-
-                    nameParts.Push(
-                        GetIndexerInvocation(
-                            binaryExpression.Right,
-                            expression.Parameters.ToArray()));
+                        builder.Insert(0, indexerInvokationResult);
 
-                    part = binaryExpression.Left;
-                }
-                else if (part.NodeType == ExpressionType.MemberAccess)
-                {
-                    var memberExpressionPart = (MemberExpression)part;
-                    var name = memberExpressionPart.Member.Name;
-
-                    // If identifier contains "__", it is "reserved for use by the implementation" and likely compiler-
-                    // or Razor-generated e.g. the name of a field in a delegate's generated class.
-                    if (name.Contains("__"))
+                        part = binaryExpression.Left;
+                    }
+                    else if (part.NodeType == ExpressionType.MemberAccess)
+                    {
+                        var memberExpressionPart = (MemberExpression)part;
+                        var name = memberExpressionPart.Member.Name;
+
+                        // If identifier contains "__", it is "reserved for use by the implementation" and likely compiler-
+                        // or Razor-generated e.g. the name of a field in a delegate's generated class.
+                        if (name.Contains("__"))
+                        {
+                            // Exit loop. Should have the entire name because previous MemberAccess has same name as the
+                            // leftmost expression node (a variable).
+                            break;
+                        }
+
+                        builder.Insert(0, name);
+                        builder.Insert(0, '.');
+                        part = memberExpressionPart.Expression;
+                    }
+                    else if (part.NodeType == ExpressionType.Parameter)
+                    {
+                        // Exit loop. Have the entire name because the parameter cannot be used as an indexer; always the
+                        // leftmost expression node.
```

Comment:
Can't parse this comment. Also why does this block exists -- does the same thing as the regular `else` block?